web/core/src/jsTest/kotlin/elements/RadioGroupTests.kt

```diff
@@ -0,0 +1,181 @@
+                RadioInput(value = Rg1.V1)
+                RadioInput(value = Rg1.V2)
+                RadioInput(value = Rg1.V3)
+            }
+        }
+
+        assertEquals(
+            "<div>" +
+                    "<input type=\"radio\" name=\"g1\" value=\"V1\">" +
+                    "<input type=\"radio\" name=\"g1\" value=\"V2\">" +
+                    "<input type=\"radio\" name=\"g1\" value=\"V3\">" +
+                    "</div>",
+            root.outerHTML
+        )
+
+        repeat(3) {
+            assertEquals(false, (root.childNodes.item(it) as HTMLInputElement).checked)
+        }
+    }
+
+    @Test
+    fun radioGroupCheckedValueChanges() = runTest {
+        var rgCheckedState by mutableStateOf<Rg1?>(null)
+
+        composition {
+            RadioGroup(checkedValue = rgCheckedState, name = "g1") {
+                RadioInput(value = Rg1.V1)
+                RadioInput(value = Rg1.V2)
+                RadioInput(value = Rg1.V3)
+            }
+        }
+
+        val expectedHtml = "<div>" +
+                "<input type=\"radio\" name=\"g1\" value=\"V1\">" +
+                "<input type=\"radio\" name=\"g1\" value=\"V2\">" +
+                "<input type=\"radio\" name=\"g1\" value=\"V3\">" +
+                "</div>"
+
+        assertEquals(expectedHtml, root.outerHTML)
+        repeat(3) {
+            assertEquals(false, (root.childNodes.item(it) as HTMLInputElement).checked)
+        }
+
+        Rg1.values().forEachIndexed { index, rg ->
+            rgCheckedState = rg
+            waitForRecompositionComplete()
+
+            assertEquals(expectedHtml, root.outerHTML)
+            repeat(3) {
+                assertEquals(it == index, (root.childNodes.item(it) as HTMLInputElement).checked)
+            }
+        }
+
+        rgCheckedState = null
+        waitForRecompositionComplete()
+
+        assertEquals(expectedHtml, root.outerHTML)
+        repeat(3) {
+            assertEquals(false, (root.childNodes.item(it) as HTMLInputElement).checked)
+        }
+    }
+
+    @Test
+    fun radioGroupWithNotDirectChildrenRadioInputsWorksCorrectly() = runTest {
+        var rgCheckedState by mutableStateOf<Rg1?>(null)
+
+        composition {
+            RadioGroup(checkedValue = rgCheckedState, name = "g1") {
+                Div { RadioInput(value = Rg1.V1) }
+                Div { RadioInput(value = Rg1.V2) }
+                Div { RadioInput(value = Rg1.V3) }
+            }
+        }
+
+        val expectedHtml = "<div>" +
+                "<div><input type=\"radio\" name=\"g1\" value=\"V1\"></div>" +
+                "<div><input type=\"radio\" name=\"g1\" value=\"V2\"></div>" +
+                "<div><input type=\"radio\" name=\"g1\" value=\"V3\"></div>" +
+                "</div>"
+
+        assertEquals(expectedHtml, root.outerHTML)
+        repeat(3) {
+            assertEquals(false, (root.childNodes.item(it)!!.firstChild as HTMLInputElement).checked)
```

Comment:
I think `nextChild()` won't help in this case, since I make assertions on the same children multiple times within the same test at different times. So I can use `nextChild()` only for the first set of assertions, but later I'll need to use `root.childNodes.item(ix)` because the iterator will not have any elements by that time.

So in this case I prefer to keep it consistent within the same test.